### DISCUSSION TITLE
Collision Events?

### DISCUSSION THREAD

**Discussion by arnicas** [Annoraaq/grid-engine#387] [Q&A] [answered]
Hi, I haven't been able to find a collision event in the lib... it looks like maybe people have written hacks to detect it? I want to be able to handle collisions differently depending on what they are: 2 NPC's collide, a player and an NPC collide, etc. Or by type of NPC. And obviously collision with a tile object like a chest.

Is there anything I've missed?
Thanks, it looks quite fun and cool!
Lynn

**Comment by Annoraaq** [comment]
Hey @arnicas,
I guess that depends a bit on what you consider to be a collision. Do you mean occupying the same tile at the same time?
If both characters do not block each other (are allowed to be at the same tile position), you could check for a collision by using:
[positionChangeFinished](https://annoraaq.github.io/grid-engine/api/classes/GridEngine.html#positionChangeFinished) and [getCharactersAt](https://annoraaq.github.io/grid-engine/api/classes/GridEngine.html#getCharactersAt).

In case they **are** blocking each other, a collision could be defined as an approach of a character to go to a tile which is already blocked by another character. In this case you can use [directionChanged](https://annoraaq.github.io/grid-engine/api/classes/GridEngine.html#directionChanged) and check for a character on the tile that was tried to enter using [getFacingPosition](https://annoraaq.github.io/grid-engine/api/classes/GridEngine.html#getFacingPosition) and [getCharactersAt](https://annoraaq.github.io/grid-engine/api/classes/GridEngine.html#getCharactersAt)

#181 and #187 could also be helpful.

**Comment by arnicas** [comment, marked as answer]
Super, thank you - I will do the latter.
